Libraries/LibWeb/DOM/Document.h

```diff
@@ -1265,6 +1276,10 @@ class Document
     HashTable<GC::Ref<Element>> m_render_blocking_elements;
 
     HashTable<WeakPtr<Node>> m_pending_nodes_for_style_invalidation_due_to_presence_of_has;
+
+    // https://www.w3.org/TR/css-properties-values-api-1/#dom-window-registeredpropertyset-slot
+    // FIXME: For now, it's the "raw" CSSPropertyRule. Later, we'll need to try to parse it into a "real" style value using the property's syntax.
+    HashMap<FlyString, Web::CSS::CSSPropertyRule*> m_registered_custom_properties;
```

Comment:
I think storing CSSPropertyRule pointers here is actually correct and you don't need the FIXME. (Correct me if I'm wrong though.)

However, just storing `CSSPropertyRule*` isn't safe. It's a garbage-collected type, which means these pointers could become invalid at some point. To make that safe, store it using a `GC::Ref<CSS::CSSPropertyRule>`, and then visit `m_registered_custom_properties` in `Document::visit_edges()` which will tell the garbage collector that someone's using it and it's not garbage.